static/app/views/insights/common/utils/useStarredSegment.tsx

```diff
@@ -6,31 +6,32 @@ import {
   addSuccessMessage,
 } from 'sentry/actionCreators/indicator';
 import {t} from 'sentry/locale';
+import type {Project} from 'sentry/types/project';
 import {useMutation} from 'sentry/utils/queryClient';
 import useApi from 'sentry/utils/useApi';
 import useOrganization from 'sentry/utils/useOrganization';
 
 type StarTransactionParams = {
-  project_id: string;
-  segment_name: string;
+  project_id?: string;
+  segment_name?: string;
 };
 
 const URL_PREFIX = '/insights/starred-segments/';
 
 interface Props {
   initialIsStarred: boolean;
-  projectSlug: string;
   segmentName: string;
+  project?: Project | undefined;
 }
 
-export function useStarredSegment({initialIsStarred, projectSlug, segmentName}: Props) {
+export function useStarredSegment({initialIsStarred, project, segmentName}: Props) {
   const [isStarred, setIsStarred] = useState(initialIsStarred);
   const organization = useOrganization();
   const api = useApi();
 
   const url = `/organizations/${organization.slug}${URL_PREFIX}`;
   const data: StarTransactionParams = {
-    project_id: projectSlug,
+    project_id: project?.id,
```

Comment:
Is the ID the only part of the project object you're using? If yes, would be better to just pass the ID, rather than the whole object! Easier to mock, more flexible for contexts where projects aren't easily available, etc.

Reply:
It makes it a bit less likely that someone passes some random string as the id, this has more of a guarantee that the id is actually associated to a project.

Reply:
Alternatively, the best of both worlds we can just check if the project exists in `useStarredSegment`?

Reply:
> less likely that someone passes some random string as the id

Is this a thing that people do!? We accept `projectId` as a prop in like a hundred components!

Reply:
Fair enough, I can change it. The backend will handle invalid stuff anyways